posts/2021-01-22-jwtbuilder-redirects-21001.adoc

```diff
@@ -0,0 +1,162 @@
++
+Previously, Open Liberty contained a flaw in the detection of changes to the JARs that compose the server implementation. When such changes were detected Liberty would force the JAR to be uninstalled and installed again. In most cases this allowed the runtime to recover and function properly. In specific cases, where some core component was re-installed, Liberty would not properly recover and it would result in some Java packages to be unavailable for class loading.  For example, on Java 11 the package `javax.xml.soap` would become unavailable to the application class loaders. 
++
+The most common environment where this occurred was running Open Liberty in OpenShift. This behavior has now been corrected for Open Liberty 21.0.0.1.
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/14936[Enable MyFaces 2.3.7 for Open Liberty]
++
+The `jsf-2.3` feature in Open Liberty makes use of link:https://myfaces.apache.org/core23/index.html[Apache MyFaces] JavaServer Faces implementation. With the release of Apache MyFaces 2.3.7 a number of improvements and bug fixes have been made, for more information visit the link:https://issues.apache.org/jira/secure/ReleaseNote.jspa?projectId=10600&version=12346524[Apache MyFaces 2.3.7 changelog].
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/15337[Dynacache initialization issue when ID is missing]
++
+An external user discovered that the initialization of a `distributedMap` fails with a NullPointerException if the `id` element is not present in the `distributedMap` configuration.
+A clearer message should have been displayed to indicate the required `id` element is missing.  To address this the `distributedMap` definition has been modified to mark the `id` element as required.
++
+The configuration runtime will now issue an error message: 
++
+`CWWKG0058E: The element distributedMap with the unique identifier default-0 is missing the required attribute id.`
++
+and the `distributedMap` will not be put into service. This new behaviour is introduced in Open Liberty 21.0.0.1. 
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/14925[OAuth user registry lookups may use incorrect custom cache key]
++
+In previous releases, a flaw existed where a previously authenticated user’s `Subject` might not be found in the authentication cache during an OAuth authentication flow, when using a custom user registry. The cache key used to retrieve user claims from the authentication cache was based on the realm and username, but the correct cache key might be a combination of the OAuth provider name and the OAuth token object itself. The behavior has been corrected and the appropriate cache key should now be used.
++
+For more information visit the link:https://openliberty.io/docs/21.0.0.1/reference/feature/oauth-2.0.html[Open Liberty Documentation].
+
+* Add HTTP/2 IOException for misbehaving client error case
```

Comment:
Link to issue missing
https://github.com/OpenLiberty/open-liberty/issues/15220

Reply:
Added.